Fix context menu indexing

kanishk98 opened the pull request:
---
<!--
Remove the fields that are not appropriate
Please include:
-->

**What's this PR do?**

Addresses a problem where the context menu would get the incorrect index
of server in sidebar and cause a crash when attempting to remove the
server.

**Any background context you want to provide?**

Discussion in [this thread](https://chat.zulip.org/#narrow/stream/16-desktop/topic/sidebar.20context.20menu). 

**You have tested this PR on:**
  - [x] Windows
  - [ ] Linux/Ubuntu
  - [ ] macOS

abhigyank reviewed (approved):
Fix works correctly.
A comment somewhere about the need of the index parameter would be nice I suppose.

kanishk98 committed:
Fix context menu indexing

Addresses a problem where the context menu would get the incorrect index
of server in sidebar and cause a crash when attempting to remove the
server.

Add comment justifying extra index parameter

akashnimare commented:
Thanks, @abhigyank, @kanishk98 🚀
